src/common/reducers/repositories.js

```diff
@@ -1,42 +1,117 @@
+import merge from 'lodash/merge';
+import union from 'lodash/union';
+
 import * as ActionTypes from '../actions/repositories';
-import { parseGitHubRepoUrl } from '../helpers/github-url';
+
+// XXX move to entities.js
+// XXX should we be handling snaps here too?
+// no, because it's not part of the normalized result from github
+// so this should really be called github_entities
+// no, decompose instead ...
+export function entities(state = { snaps: {}, repos: {} }, action) {
+  if (action.payload && action.payload.entities) {
+    return merge({}, state, action.payload.entities);
+  }
+
+
+  const reposInitialState = {
+    __isSelected: false,
```

Comment:
To be fair, it's unlikely that isSelected is every likely to be clobbered by gh, but a namespace of some kind does tell you that you're dealing with a property sourced from the client rather than a 3rd party source.

Where I was going with this earlier on, and maybe I should have stuck at it, was another object keyed on id, but for client state of entities.